ibm/service/eventstreams/data_source_ibm_event_streams_schema_global_compatibility_rule.go

```diff
@@ -0,0 +1,118 @@
+			},
+		},
+	}
+}
+
+// read global compatibility rule properties using the schema registry API
+func dataSourceIBMEventStreamsSchemaGlobalCompatibilityRuleRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	schemaregistryClient, err := meta.(conns.ClientSession).ESschemaRegistrySession()
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, "Error getting Event Streams schema registry session", "ibm_event_streams_schema_global_rule", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+	adminURL, instanceCRN, err := getSchemaRuleInstanceURL(d, meta)
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, "Error getting Event Streams schema registry URL", "ibm_event_streams_schema_global_rule", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+	schemaregistryClient.SetServiceURL(adminURL)
+
+	getOpts := &schemaregistryv1.GetGlobalRuleOptions{}
+	getOpts.SetRule(schemaCompatibilityRuleType)
+	rule, _, err := schemaregistryClient.GetGlobalRuleWithContext(context, getOpts)
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, "GetGlobalRule returned error", "ibm_event_streams_schema_global_rule", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+	if rule.Config == nil {
+		tfErr := flex.TerraformErrorf(err, "Unexpected nil config when getting global compatibility rule", "ibm_event_streams_schema_global_rule", "read")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+	d.SetId(getSchemaGlobalCompatibilityRuleID(instanceCRN))
+	d.Set("resource_instance_id", instanceCRN)
+	d.Set("config", *rule.Config)
+	return nil
+}
+
+func getSchemaRuleInstanceURL(d *schema.ResourceData, meta interface{}) (string, string, error) {
+	instanceCRN := d.Get("resource_instance_id").(string)
+	if instanceCRN == "" { // importing
+		id := d.Id()
+		crnSegments := strings.Split(id, ":")
+		if len(crnSegments) != 10 || crnSegments[8] != schemaGlobalCompatibilityRuleResourceType {
+			return "", "", fmt.Errorf("ID '%s' is not a schema global compatibility resource", id)
+		}
+		crnSegments[8] = ""
+		crnSegments[9] = ""
+		instanceCRN = strings.Join(crnSegments, ":")
+		d.Set("resource_instance_id", instanceCRN)
+	}
+
+	instance, err := getInstanceDetails(instanceCRN, meta)
+	if err != nil {
+		return "", "", err
+	}
+	adminURL := instance.Extensions["kafka_http_url"].(string)
+	planID := *instance.ResourcePlanID
+	valid := strings.Contains(planID, "enterprise")
+	if !valid {
+		return "", "", fmt.Errorf("schema registry is not supported by the Event Streams %s plan, enterprise plan is expected",
+			planID)
+	}
+	return adminURL, instanceCRN, nil
+}
+
+func getSchemaGlobalCompatibilityRuleID(instanceCRN string) string {
+	crnSegments := strings.Split(instanceCRN, ":")
+	crnSegments[8] = schemaGlobalCompatibilityRuleResourceType
+	crnSegments[9] = ""
```

Comment:
Our convention in Event Streams is that the first 8 fields of the CRN identify the service-instance, and the last two identify the resource on the instance.  `crnSegment[8]` is the type of the resource, and `crnSegment[9]` is the identifier for the resource.

The type for the global rule is `schema-global-compatibility-rule`.  Since the global compatibility rule is global, there is only one instance, so it does not have a unique identifier field. So we set segment 9 to `""`.

Example:
- input instanceCRN is `crn:v1:staging:public:messagehub-vnext-integration:eu-gb:a/6db1b0d0b5c54ee5c201552547febcd8:4088c6f5-6a6b-49e8-a31d-712f09f23058::`
- global rule CRN will be `crn:v1:staging:public:messagehub-vnext-integration:eu-gb:a/6db1b0d0b5c54ee5c201552547febcd8:4088c6f5-6a6b-49e8-a31d-712f09f23058:schema-global-compatibility-rule:`

In practice, the split segment 9 should already be `""` since it _should_ be coming from a service-instance CRN, in which the last two fields are not set. Setting it to `""` is a precaution in case some other value was provided.